sys/include/div.h

```diff
@@ -0,0 +1,16 @@
+#ifndef DIV_H
+#define DIV_H
+
+#include <stdint.h>
+
+static inline uint64_t div_u64_by_1000000(uint64_t val)
+{
+    return (uint64_t)(val * 0x431bde83) >> (0x12 + 32);
```

Comment:
I have been thinking about this and #3990, I think the best solution may be
to convert the xtimer to use timer ticks internally like this PR, but to
use an opaque struct type to make the compiler error out mismatched timer
ticks vs microsecond constants.
I can't prepare a PR right now but after testing and working with #3990 for
some time that PR is beginning to feel unnecessarily convoluted and too
much overhead..